### PR TITLE
zathura: make compatible with darwin

### PR DESCRIPTION
**Pull request by pallavagarwal07**
###### Motivation for this change
I wanted to use zathura on Mac.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by LnL7** [comment]
@GrahamcOfBorg build zathura

File: pkgs/applications/misc/zathura/pdf-mupdf/default.nix [hunk 1]
```diff
@@ -11,7 +12,18 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ zathura_core gtk girara openssl mupdf libjpeg jbig2dec openjpeg ];
+
+  buildInputs = [
+    zathura_core gtk girara openssl mupdf libjpeg jbig2dec openjpeg
+    gtk-mac-integration
```
Comment (LnL7) [inline pkgs/applications/misc/zathura/pdf-mupdf/default.nix]: Looks like you missed an optional

**Review by LnL7 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/zathura/djvu/default.nix [hunk 2]
```diff
@@ -13,6 +13,13 @@ stdenv.mkDerivation rec {
 
   patches = [ ./gtkflags.patch ];
 
+  patchPhase = stdenv.lib.optionalString stdenv.isDarwin ''
```
Comment (LnL7) [inline pkgs/applications/misc/zathura/djvu/default.nix]: Is the patchPhase override here intentional? This disables the gtkflags.patch, either way it's better to make that explicit and use `prePatch`/`postPatch` for extra substitutions like this.

**Comment by pallavagarwal07** [comment]
@LnL7  Done

**Comment by Mic92** [comment]
@GrahamcOfBorg build zathura

**Pull request merged**